### PR TITLE
Update python scripts for lgtm

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

**Comment by evverx** [comment]
This pull request **introduces 2 alerts** and **fixes 4** when merging fa80691eb5b40c97445e4c1237dc710140f9bcda into c90ee83400783739ac6df9f12c43bd7074dea8fa - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-d29c7b038c10ecd271cc5d271a08f31c122896ce)

**new alerts:**

* 2 for Syntax error

**fixed alerts:**

* 2 for Encoding error
* 1 for Unused local variable
* 1 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by yuwata** [comment]
@evverx Why the python scripts are parsed as python2?
```
invalid syntax (in Python 2.7).
```

**Comment by evverx** [comment]
@yuwata to be honest, I didn't know LGTM started to analyze the python code and at the moment I didn't know what to say :-) I found a similar question at https://discuss.lgtm.com/t/python-3-detection/1153 so probably we ran into the same issue. Let's cc @samlanning just in case.

**Comment by yuwata** [comment]
@evverx I see. Thanks. I thought you enabled the python code analysis.

File: .ycm_extra_conf.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env python
-
+# -*- Mode: python; coding: utf-8; indent-tabs-mode: nil -*-
```
Comment (poettering) [inline .ycm_extra_conf.py]: hmm, this is for emacs, no? but why would emacs ever assume something else than utf-8?

a while back we dropped all emacs mode-lines from the project, in favour of dir-locals. Why not handle this the same way?

File: tools/catalog-report.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ def log_entry(x):
 
 for i, x in enumerate(j):
     if i % freq == 0:
-        print(i, end='\r')
+        print('{}'.format(i), end='\r')
```
Comment (poettering) [inline tools/catalog-report.py]: i guess i am not a python guru enough for this, but what precisely does this do? afaics this just changes the type of what is printed here from an integer to a string, but if that's the intention, why not use str()?

File: .ycm_extra_conf.py (hunk 1 above)
Comment (poettering) [reply .ycm_extra_conf.py]: I understand now that this is actually not for emacs but for python. However it appears to me this is unnecessary for python3:

https://www.python.org/dev/peps/pep-3120/

Hence I figure this commit can be dropped?

**Comment by evverx** [comment]
This pull request **fixes 6 alerts** when merging d8a0bcfd77832357f7db43e072a7ff783dffd369 into a01eb97b6c9b43a4303c55cf10b8ab469300f343 - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-bdfa0bb306bf7a9b60c8c2df0bd1011ab5f5fef2)

**fixed alerts:**

* 2 for Syntax error
* 2 for Encoding error
* 1 for Unused local variable
* 1 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by samlanning** [comment]
Yep, looks like you sorted it with the `lgtm.yml` file :)

File: .ycm_extra_conf.py (hunk 1 above)
Comment (yuwata) [reply .ycm_extra_conf.py]: Yeah, dropped.

**Comment by yuwata** [comment]
@poettering I've dropped weird 'fixes'. PTAL.

**Comment by poettering** [comment]
CI failures look unrelated.

**Pull request merged**